src/lib.rs

```diff
@@ -1444,7 +1444,13 @@ impl Builder {
     }
 
     /// Generate the Rust bindings using the options built up thus far.
-    pub fn generate(mut self) -> Result<Bindings, ()> {
+    #[deprecated(since = "0.59.3", note = "please use `gen` instead")]
+    pub fn generate(self) -> Result<Bindings, ()> {
+        self.gen().map_err(|_| ())
+    }
+
+    /// Generate the Rust bindings using the options built up thus far, or `Err` on failure.
+    pub fn gen(mut self) -> Result<Bindings, BindgenError> {
```

Comment:
I suspect most uses of `Builder::generate` will just `.unwrap()` directly, this is what all the documentation recommends, and you usually don't care much about this type of error handling when making build scripts.

I think it's unfortunate that we resort to using a less descriptive function name here, and also, it will be something that _every_ user of `bindgen` will have to change, for no real benefit (again, assuming they use `.unwrap`).

So I propose we just change the original function signature in a major release (unfortunately without a deprecation period).

Reply:
In [`tests/tests.rs:203`](https://github.com/rust-lang/rust-bindgen/blob/89032649044d875983a851fff6fbde2d4e2ceaeb/tests/tests.rs#L203), the code running `generate` depends that it returns `()` on failure originally. We should think that someone is doing the same. So I think this is a breaking change.

Reply:
Yeah, it definitely is a breaking change, no doubt, and it should be noted as such in the changelog, and can only included in a major release (would currently be `0.60.0`).

But I don't think it is a big enough breaking change to warrant a deprecation period, considering the downsides.

Reply:
Hmm, I think we shouldn't introduce the breaking change immediately but sounds good about just changing the return value. Also the version is now `0.x.y` so it's ok to do. I will try.